ownCloud/Settings/LogFilesViewController.swift

```diff
@@ -0,0 +1,235 @@
+			}
+		}
+
+		return cell!
+	}
+
+	override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		tableView.deselectRow(at: indexPath, animated: true)
+	}
+
+	override func tableView(_ tableView: UITableView, editActionsForRowAt indexPath: IndexPath) -> [UITableViewRowAction]? {
+
+		return [
+			UITableViewRowAction(style: .destructive, title: "Delete".localized, handler: { [weak self] (_, indexPath) in
+				self?.removeLogRecord(at: indexPath)
+			})
+		]
+	}
+
+	// MARK: - Private Helpers
+
+	private func populateLogFileList() {
+		guard let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter else { return }
+		self.logRecords = logFileWriter.logRecords().sorted(by: { (record1, record2) in
+			if let date1 = record1.creationDate, let date2 = record2.creationDate {
+				return date1 > date2
+			}
+			return false
+		})
+		self.tableView.reloadData()
+	}
+
+	private func shareLogRecord(at indexPath:IndexPath) {
+
+		let logRecord = self.logRecords[indexPath.row]
+		let shareableFileName = logRecord.name + ".txt"
+		let shareableLogURL = FileManager.default.temporaryDirectory.appendingPathComponent(shareableFileName)
+
+		do {
+			if FileManager.default.fileExists(atPath: shareableLogURL.path) {
+				try FileManager.default.removeItem(at: shareableLogURL)
+			}
+
+			try FileManager.default.copyItem(atPath: logRecord.url.path, toPath: shareableLogURL.path)
+		} catch {
+		}
+
+		let shareViewController = UIActivityViewController(activityItems: [shareableLogURL], applicationActivities:nil)
+		shareViewController.completionWithItemsHandler = { (_, _, _, _) in
+			do {
+				try FileManager.default.removeItem(at: shareableLogURL)
+			} catch {
+			}
+		}
+
+		if UIDevice.current.isIpad() {
+			shareViewController.popoverPresentationController?.sourceView = self.view
+		}
+		self.present(shareViewController, animated: true, completion: nil)
+	}
+
+	private func removeLogRecord(at indexPath:IndexPath) {
+
+		guard let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter else { return }
+
+		let record = self.logRecords[indexPath.row]
+		logFileWriter.deleteLogRecord(record)
+
+		self.logRecords.remove(at: indexPath.row)
+		self.tableView.deleteRows(at: [indexPath], with: .automatic)
+	}
+
+	@objc private func removeAllLogs() {
+		let alert = UIAlertController(with: "Delete all log files?".localized,
+									  message: "This action can't be undone.".localized,
+									  destructiveLabel: "Delete all".localized,
+									  preferredStyle: .alert,
+									  destructiveAction: {
+			OCLogger.shared.pauseWriters(intermittentBlock: {
+				if let logFileWriter = OCLogger.shared.writer(withIdentifier: .writerFile) as? OCLogFileWriter {
+					logFileWriter.cleanUpLogs(true)
+				}
+			})
+		})
+
+		self.present(alert, animated: true, completion: nil)
+	}
+
+	@objc private func handleLogRotationNotification() {
+		populateLogFileList()
```

Comment:
Just noticed during testing: this notification can be delivered on a background thread and then lead to `tableView.reloadData()` being called on a thread different than the main thread.
```suggestion
		OnMainThread {
			self.populateLogFileList()
		}
```